arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -999,6 +1007,9 @@ Result IResearchDataStore::commitUnsafeImpl(
       _lastCommittedTickOne = lastTickBeforeCommitOne;
       _lastCommittedTickTwo = lastTickBeforeCommitOne;
       impl.tick(_lastCommittedTickOne);
+      auto reader = _dataStore.loadSnapshot()->_reader;
+      _dataStore.storeSnapshot(
+          std::make_shared<DataSnapshot>(std::move(reader), engineSnapshot));
```

Comment:
If this throws an exception, are things then left in a bad state (e.g. `_lastCommittedTickTwo` was already advanced before)?

Reply:
Worst case - we will have old snapshot in the link. Will try to update it next time. Index will still be consistent, just more "eventual"